### PR TITLE
Enable selecting multiple filter items

### PR DESCRIPTION
**Pull request by norbye**
# Description

Enables users to check multiple items from the filter dropdowns

Requires https://github.com/webkom/lego/pull/3518 to work

# Result

https://github.com/webkom/lego-webapp/assets/13599770/6b27b6a9-6ffe-41be-993c-0458255706bc

# Testing

- [ ] I have thoroughly tested my changes.

Will do a bit more testing before merging this, but this should mainly be it

---

Resolves ABA-366

### CODE REVIEW

**Review by danielyanghansen — APPROVED** [review]

**Review by ivarnakken — APPROVED** [review]
🏅

File: app/components/Table/index.tsx [hunk 1]
```diff
@@ -81,17 +84,14 @@ export default class Table extends Component<Props, State> {
...
-  };
   toggleSearch = (dataIndex: string) => {
     this.setState({
       isShown: {
         [dataIndex]: !this.state.isShown[dataIndex],
       },
     });
   };
-  toggleFilter = (dataIndex: string) => {
+  toggleIsShown = (dataIndex: string) => {
```
[the remark this answers is not in the document]
Comment (norbye) [reply app/components/Table/index.tsx:94]: Hmm yeah then the naming makes a little more sense - although it's still stoopid..

Wanted to rewrite it to a functional component and do some cleanup, but I quickly realized that would include quite a few changes and make these changes drown in the review. So I figured I'd rather get this in first and then fix up what was already there and rewrite it in the next PR

[the remark this answers is not in the document]
Comment (eikhr) [reply app/components/Table/index.tsx:94]: Would love to get #4259 merged too before you do any major refactoring

**Review by eikhr — APPROVED** [review]
~If i understand this correctly, this changes all filters to be arrays. I am a little bit worried that some of the tables that use filters for query parameters might break by this. I know the email ones do, unsure if there any others. If you haven't already, you should tests this:)~ 
~Edit: nevermind, just tested it and it works fine!~
Edit: No, I tested some more and it only works for some fields. I realize now, this is what your backend PR adresses, but there are more fields to fix. Committee membership and year of study filters don't work in email user list if you select multiple values. 

Other than that, it looks great!

File: app/components/Table/index.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (norbye) [reply app/components/Table/index.tsx:94]: absolutely, don't think I'll have the time to fix it for a little while so that shouldn't be a problem

**Comment by norbye** [comment]
> Edit: No, I tested some more and it only works for some fields. I realize now, this is what your backend PR adresses, but there are more fields to fix. Committee membership and year of study filters don't work in email user list if you select multiple values.

Ahh yeah, thanks for doing some of the testing for me! Will go through it and update the backend filtering for the other fields that require it before I finish this

**Comment by norbye** [comment]
Resolved in #4411

**Pull request closed without merging**